spring-rabbit/src/main/java/org/springframework/amqp/rabbit/annotation/MultiRabbitListenerAnnotationBeanPostProcessor.java

```diff
@@ -71,13 +73,33 @@ private RabbitListener proxyIfAdminNotPresent(final RabbitListener rabbitListene
 	 */
 	protected String resolveMultiRabbitAdminName(RabbitListener rabbitListener) {
 		String admin = super.resolveExpressionAsString(rabbitListener.admin(), "admin");
-		if (!StringUtils.hasText(admin) && StringUtils.hasText(rabbitListener.containerFactory())) {
-			admin = rabbitListener.containerFactory() + RabbitListenerConfigUtils.MULTI_RABBIT_ADMIN_SUFFIX;
+		if (StringUtils.hasText(admin)) {
+
+			return admin;
 		}
-		if (!StringUtils.hasText(admin)) {
-			admin = RabbitListenerConfigUtils.RABBIT_ADMIN_BEAN_NAME;
+
+		if (!StringUtils.hasText(rabbitListener.containerFactory())) {
+
+			return RabbitListenerConfigUtils.RABBIT_ADMIN_BEAN_NAME;
 		}
-		return admin;
+
+		if (this.beanFactory instanceof ConfigurableListableBeanFactory clbf) {
+
+			Object resolved = super.resolveExpression(rabbitListener.containerFactory());
+			if (resolved instanceof RabbitListenerContainerFactory<?> rlcf) {
+
+				var containerFactoryByName = clbf.getBeansOfType(RabbitListenerContainerFactory.class);
```

Comment:
I still don't approve this.
See `BeanNameAware` to be implemented on the `RabbitListenerContainerFactory`.
Together with a getter.
The `RabbitAdmin` already comes with the impl.
That `super.resolveExpressionAsString(rabbitListener.admin(), "admin")` would fail if an expression points to the target bean.
So, let's improve that part as well!

Reply:
@artembilan 

I implement `BeanNameAware`. Please review and tell me if I miss something